arch/riscv/Kconfig

```diff
@@ -100,6 +100,21 @@ config RISCV_SOC_HAS_CUSTOM_SYS_IO
 	  the RISC-V SoC needs to do something different and more than reading and
 	  writing the registers.
 
+config RISCV_SOC_HAS_CUSTOM_SMP_IPI
+	bool
+	select SCHED_IPI_SUPPORTED
+	help
+	  Hidden option to allow SoC to overwrite SMP IPI implementation
+	  Enable this hidden option and do the following:
+	  - register a custom SW IRQ handler
+	  - implement arch_sched_ipi()
+	  - implement z_riscv_flush_fpu_ipi()
+
+config RISCV_SOC_HAS_CUSTOM_PER_CORE_INIT
+	bool
+	help
+	  Hidden option to allow SoC to specify custom per-core initialization
```

Comment:
should we specify here that the user needs to implement `z_soc_per_core_init` as well?

Reply:
Updated to reflect your comments; please re-review

Reply:
@carlocaione  - please re-review